File: VB/LinqWithEditing/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.Data.Linq
Imports DevExpress.XtraGrid.Views.Base

Namespace LinqWithEditing
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private dc As NorthwindProductsDataContext

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			BindGrid()
		End Sub

		Private Sub BindGrid()
			dc = New NorthwindProductsDataContext()
			Dim categories As Table(Of Category) = dc.GetTable(Of Category)()
			categoryBindingSource.DataSource = categories
		End Sub
		Private Sub gridControl1_EmbeddedNavigator_ButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.NavigatorButtonClickEventArgs) Handles gridControl1.EmbeddedNavigator.ButtonClick
			If e.Button.ButtonType = DevExpress.XtraEditors.NavigatorButtonType.EndEdit Then
				Dim view As ColumnView = CType(gridControl1.FocusedView, ColumnView)
				view.CloseEditor()
				If view.UpdateCurrentRow() Then
					Try
						dc.SubmitChanges(ConflictMode.FailOnFirstConflict)
					Catch ex As ChangeConflictException
						MessageBox.Show(ex.Message)
						For Each occ As ObjectChangeConflict In dc.ChangeConflicts
							occ.Resolve(RefreshMode.OverwriteCurrentValues)
						Next occ
						'view.RefreshData();
					Catch ex As Exception
						MessageBox.Show(ex.Message)
						BindGrid()
					End Try
				End If
				e.Handled = True
			End If
		End Sub
	End Class
End Namespace

File: CS/LinqWithEditing/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Data.Linq;
using DevExpress.XtraGrid.Views.Base;

namespace LinqWithEditing {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }

        NorthwindProductsDataContext dc;

        private void Form1_Load(object sender, EventArgs e) {
            BindGrid();
        }

        private void BindGrid() {
            dc = new NorthwindProductsDataContext();
            Table<Category> categories = dc.GetTable<Category>();
            categoryBindingSource.DataSource = categories;
        }
        private void gridControl1_EmbeddedNavigator_ButtonClick(object sender, DevExpress.XtraEditors.NavigatorButtonClickEventArgs e) {
            if(e.Button.ButtonType == DevExpress.XtraEditors.NavigatorButtonType.EndEdit) {
                ColumnView view = (ColumnView)gridControl1.FocusedView;
                view.CloseEditor();
                if(view.UpdateCurrentRow()) {
                    try {
                        dc.SubmitChanges(ConflictMode.FailOnFirstConflict);
                    }
                    catch(ChangeConflictException ex) {
                        MessageBox.Show(ex.Message);
                        foreach(ObjectChangeConflict occ in dc.ChangeConflicts) {
                            occ.Resolve(RefreshMode.OverwriteCurrentValues);
                        }
                        //view.RefreshData();
                    }
                    catch(Exception ex) {
                        MessageBox.Show(ex.Message);
                        BindGrid();
                    }
                }
                e.Handled = true;
            }
        }
    }
}
